### ISSUE TITLE
[AngularFirePerfomance] Manual trace is not working

### ISSUE THREAD

**Issue by jmaelstafbetclic** [angular/angularfire#2518]
### Version info

**Angular:** 8.2.13

**Firebase:** 7.15.4

**AngularFire:** 5.4.2

**Other (e.g. Ionic/Cordova, Node, browser, operating system):** Chrome 83.0.4103.97, OSX 10.15.5

### How to reproduce these conditions

**Failing test unit, Stackblitz demonstrating the problem**

https://stackblitz.com/edit/angular-fire-start-e8e6cg?file=app%2Fapp.component.ts

**Steps to set up and reproduce**

Follow the doc : innject AngularFirePerfomance, create a trace => error

**Sample data and security rules**

### Debug output

** Errors in the JavaScript console **

```
preview-e005813feabe81a85b33b.js:1 ERROR TypeError: trace.start is not a function
    at AppComponent.ngOnInit (app.component.ts:21)
    at checkAndUpdateDirectiveInline (provider.ts:217)
    at checkAndUpdateNodeInline (view.ts:440)
    at checkAndUpdateNode (view.ts:400)
    at debugCheckAndUpdateNode (services.ts:433)
    at debugCheckDirectivesFn (services.ts:394)
    at Object.eval [as updateDirectives] (AppComponent_Host.ngfactory.js? [sm]:1)
    at Object.debugUpdateDirectives [as updateDirectives] (services.ts:388)
    at checkAndUpdateView (view.ts:370)
    at callWithDebugContext (services.ts:638)
console.<computed> @ preview-e005813feabe81a85b33b.js:1
preview-e005813feabe81a85b33b.js:1 Unable to display error. Open your browser's console to view.
console.<computed> @ preview-e005813feabe81a85b33b.js:1
preview-e005813feabe81a85b33b.js:1 ERROR CONTEXT DebugContext_
console.<computed> @ preview-e005813feabe81a85b33b.js:1
preview-e005813feabe81a85b33b.js:1 Angular is running in the development mode. Call enableProdMode() to enable the production mode.
preview-e005813feabe81a85b33b.js:1 ERROR Error: Uncaught (in promise): TypeError: app.performance is not a function
TypeError: app.performance is not a function
    at eval (performance.ts:34)
    at ZoneDelegate.invoke (zone.js:386)
    at Zone.run (zone.js:143)
    at NgZone.runOutsideAngular (ng_zone.ts:227)
    at MapSubscriber.eval [as project] (performance.ts:33)
    at MapSubscriber._next (map.ts:84)
    at MapSubscriber.Subscriber.next (Subscriber.ts:99)
    at SwitchMapSubscriber.notifyNext (switchMap.ts:172)
    at InnerSubscriber._next (InnerSubscriber.ts:17)
    at InnerSubscriber.Subscriber.next (Subscriber.ts:99)
    at resolvePromise (zone.js:832)
    at resolvePromise (zone.js:784)
    at eval (zone.js:894)
    at ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (ng_zone.ts:293)
    at ZoneDelegate.invokeTask (zone.js:420)
    at Zone.runTask (zone.js:188)
    at drainMicroTaskQueue (zone.js:601)
console.<computed> @ preview-e005813feabe81a85b33b.js:1
2preview-e005813feabe81a85b33b.js:1 Unhandled Promise rejection: app.performance is not a function ; Zone: <root> ; Task: Promise.then ; Value: TypeError: app.performance is not a function
    at eval (performance.ts:34)
    at ZoneDelegate.invoke (zone.js:386)
    at Zone.run (zone.js:143)
    at NgZone.runOutsideAngular (ng_zone.ts:227)
    at MapSubscriber.eval [as project] (performance.ts:33)
    at MapSubscriber._next (map.ts:84)
    at MapSubscriber.Subscriber.next (Subscriber.ts:99)
    at SwitchMapSubscriber.notifyNext (switchMap.ts:172)
    at InnerSubscriber._next (InnerSubscriber.ts:17)
    at InnerSubscriber.Subscriber.next (Subscriber.ts:99) TypeError: app.performance is not a function
    at eval (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/@angular/fire@6.0.0/bundles/angular-fire-performance.umd.js:41:45)
    at ZoneDelegate.invoke (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/zone.js@0.10.3/dist/zone.js:386:30)
    at Zone.run (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/zone.js@0.10.3/dist/zone.js:143:47)
    at NgZone.runOutsideAngular (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/@angular/core@9.1.3/bundles/core.umd.js:28315:32)
    at MapSubscriber.eval [as project] (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/@angular/fire@6.0.0/bundles/angular-fire-performance.umd.js:39:29)
    at MapSubscriber._next (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/rxjs@6.5.5/internal/operators/map.js:49:35)
    at MapSubscriber.Subscriber.next (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/rxjs@6.5.5/internal/Subscriber.js:66:18)
    at SwitchMapSubscriber.notifyNext (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/rxjs@6.5.5/internal/operators/switchMap.js:89:26)
    at InnerSubscriber._next (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/rxjs@6.5.5/internal/InnerSubscriber.js:28:21)
    at InnerSubscriber.Subscriber.next (https://angular-fire-start-e8e6cg.stackblitz.io/turbo_modules/rxjs@6.5.5/internal/Subscriber.js:66:18)
```

** Output from `firebase.database().enableLogging(true);` **

** Screenshots **
![Uploading image.png…]()

**Comment by jamesdaniels** [comment]
Looks like our docs are a bit out of date. I haven't enabled chaining on the Promise proxy in v6 yet, so creating a trace returns a raw Promise right now. You can just await it:

```ts
const trace = await this.performance.trace('some-trace');
trace.start();
trace.stop();
```

**Comment by jmaelstafbetclic** [comment]
Doing this generate another error : "TypeError: app.performance is not a function"

![image](https://user-images.githubusercontent.com/47979043/85728377-24934f80-b6f8-11ea-829e-c98696e29598.png)

**Comment by jamesdaniels** [comment]
I've been experiencing this recently too. It seems to be due to both changes in the Firebase JS SDK and overly aggressive tree-shaking. A work-around is coming together for `6.0.3`, you can try the canary build `6.0.3-canary.7cb6c03` and see if that helps.

**Comment by jamesdaniels** [comment]
Should have been addressed in `6.0.3`
